Bind SET_POLYGON_STIPPLE_PATTERN

cornytrace opened the pull request:
Causes MGS4 to boot a little bit further.
MGS4 now actually does some stuff, mostly filesystem related, and then softlocks(progress! :P). No display output on any backend unfortunately. Doesn't seem to respond to inputs either.
The information was gotten from the MESA implementation of nv30/40.
Sorry for the small commit, I am still learning the architecture of both the PS3 and the emulator.
Am I correct in using bind_array here instead of bind_range? Can anyone explain the difference?

raven02 commented:
Cool . Btw, please fix the extra spacing .

cornytrace commented on rpcs3/Emu/RSX/rsx_methods.cpp:
```diff
@@ -1228,6 +1228,7 @@ namespace rsx
 		bind_array<NV4097_SET_TEXTURE_CONTROL2, 1, 16, nullptr>();
 		bind_array<NV4097_SET_TEX_COORD_CONTROL, 1, 10, nullptr>();
 		bind_array<NV4097_SET_TRANSFORM_PROGRAM, 1, 32, nullptr>();
+        bind_array<NV4097_SET_POLYGON_STIPPLE_PATTERN, 1, 32 * 4, nullptr>();
```
Fixed, MSVC was converting my tabs to spaces.

cornytrace commented:
Fixed, MSVC was converting my tabs to spaces.

Ronaldo-brazil commented:
Why recompiler mode llvm does not work with intel (skylake) processors and when this will be fixed

kd-11 commented:
@cornytrace `bind_array<NV4097_SET_POLYGON_STIPPLE_PATTERN, 1, 32 * 4, nullptr>();` will loop over the range [NV4097_SET_POLYGON_STIPPLE_PATTERN,  NV4097_SET_POLYGON_STIPPLE_PATTERN+(32*4)] and set all the method pointers to zero which is not what you want in this case; or at least that's what I get from looking at the code.

cornytrace commented:
It is so that the pointers do not point to invalid_method, which crashes the RSX.
So yeah it implements nothing, it just stubs it out, just as the lines above and below it.

kd-11 commented:
In that case I think count should be 32 not 32*4. POLYGON_STIPPLE_PATTERN has an address range of 32 (520h to 540h; see gcm_enums.h) and the implementation of bind_array does a for loop setting the next count methods to the implementation.

Nekotekina commented:
Travis should be OK now.

cornytrace commented:
@kd-11 Yeah, I thought so too, so I first set it to 32, but still got errors for addresses under 540h, so I thought there was some pointer-size math I had to do. I changed it to 32*4 and the errors went away.
Just now, I changed it back to 32... and it worked T_T
Sorry for wasting your time!
I'll rebase while I'm at it.
